LibWeb: Split the qualified name if there is a prefix

ananas-dev opened the pull request:
This fixes (at least) one WPT test and I imported it :)

The expected behavior is described here in the spec:
https://dom.spec.whatwg.org/#concept-element-local-name

It was already implemented correctly when constructing a `QualifiedName` in `Element::validate_and_extract`.

I don't know if my sequence of string conversions is the most efficient though, but don't hesitate to comment if there is a better one so that I can learn my way around AK.

tcl3 commented on Tests/LibWeb/Text/expected/wpt-import/dom/nodes/Element-firstElementChild-namespace-xhtml.txt:
```diff
@@ -0,0 +1,11 @@
+Summary
+
+Harness status: OK
+
+Rerun
+
+Found 1 tests
+
+1 Pass
+Details
+Result	Test Name	MessagePass	firstElementChild with namespaces	
```
This test will need to be rebaselined once the conflict is resolved.

tcl3 commented on Libraries/LibWeb/XML/XMLDocumentBuilder.cpp:
```diff
@@ -113,7 +113,18 @@ void XMLDocumentBuilder::element_start(const XML::Name& name, HashMap<XML::Name,
     }
 
-    auto node = DOM::create_element(m_document, MUST(FlyString::from_deprecated_fly_string(name)), m_namespace).release_value_but_fixme_should_propagate_errors();
+    auto qualified_name = MUST(FlyString::from_deprecated_fly_string(name));
+
+    Optional<FlyString> prefix = {};
+    auto local_name = qualified_name;
+
+    if (qualified_name.bytes_as_string_view().contains(':')) {
+        auto parts = qualified_name.bytes_as_string_view().split_view(':');
+        prefix = MUST(FlyString::from_utf8(parts[0]));
+        local_name = MUST(FlyString::from_utf8(parts[1]));
+    }
```
Would it be possible to use DOM::validate_and_extract() to do this instead? If an error is returned you can set `m_has_error = true` and return as above.

ananas-dev commented:
> Thanks for working on this! Sorry it's taken a while to get to. There is currently a conflict that needs to be resolved.

Thanks for the review :)
I changed the code to use `DOM::validate_and_extract` and added the error handling you suggested. I also added the same error handling to the `DOM::create_element` call right below since I figured I would make sense.

tcl3 commented on Libraries/LibWeb/XML/XMLDocumentBuilder.cpp:
```diff
@@ -99,7 +99,23 @@ void XMLDocumentBuilder::element_start(const XML::Name& name, HashMap<XML::Name,
     }
 
-    auto node = DOM::create_element(m_document, MUST(FlyString::from_deprecated_fly_string(name)), m_namespace).release_value_but_fixme_should_propagate_errors();
+    auto qualified_name_or_err = DOM::validate_and_extract(m_document->realm(), m_namespace, MUST(FlyString::from_deprecated_fly_string(name)));
```
We use unabbreviated variable names wherever possible, so `qualified_name_or_error` would be better in this case. The same applies to `node_or_err` below.

ananas-dev replied:
Thanks for the insight, I pushed a new version with updated variable names

tcl3 reviewed (approved):
Looks good! Please use the "Resolve conversation" button when you've dealt with review feedback.
